engine/src/main/java/com/redhat/cloud/notifications/exports/ExportsEventListener.java

```diff
@@ -0,0 +1,335 @@
+
+            this.meterRegistry.counter(EXPORTS_SERVICE_FAILURES_COUNTER).increment();
+
+            final ExportError exportError = new ExportError(HttpStatus.SC_BAD_REQUEST, "the 'to' date cannot be lower than the 'from' date");
+            this.exportsService.notifyErrorExport(this.exportServicePsk, exportRequestUuid, application, resourceUuid, exportError);
+
+            return message.ack();
+        }
+
+        // Handle exporting the requested resource type.
+        if (RESOURCE_TYPE_EVENTS.equals(resource)) {
+            this.statelessSessionFactory.withSession(session -> {
+                final List<Event> events = this.eventRepository.findEventsToExport(orgId, from, to);
+
+                final ResultsTransformer<Event> resultsTransformer;
+                final String contents;
+                try {
+                    switch (format) {
+                        case CSV -> {
+                            resultsTransformer = new CSVEventTransformer();
+                            contents = resultsTransformer.transform(events);
+                            this.exportsService.uploadCSVExport(this.exportServicePsk, exportRequestUuid, application, resourceUuid, contents);
+                        }
+                        case JSON -> {
+                            resultsTransformer = new JSONEventTransformer();
+                            contents = resultsTransformer.transform(events);
+                            this.exportsService.uploadJSONExport(this.exportServicePsk, exportRequestUuid, application, resourceUuid, contents);
+                        }
+                        default -> {
+                            final ExportError exportError = new ExportError(
+                                HttpStatus.SC_BAD_REQUEST,
+                                String.format("the specified format '%s' is unsupported for the request", format)
+                            );
+                            this.exportsService.notifyErrorExport(this.exportServicePsk, exportRequestUuid, application, resourceUuid, exportError);
+                        }
+                    }
+                } catch (final TransformationException e) {
+                    Log.errorf("[export_request_uuid: %s][resource_uuid: %s][requested_format: %s] unable to transform events to the requested format: %s", exportRequestUuid, resourceUuid, format, e.getCause().getMessage(), e);
+
+                    this.meterRegistry.counter(EXPORTS_SERVICE_FAILURES_COUNTER).increment();
+
+                    final ExportError exportError = new ExportError(HttpStatus.SC_INTERNAL_SERVER_ERROR, "unable to serialize message in the correct format");
+                    this.exportsService.notifyErrorExport(this.exportServicePsk, exportRequestUuid, application, resourceUuid, exportError);
+                }
+            });
+        }
+
+        this.meterRegistry.counter(EXPORTS_SERVICE_SUCCESSES_COUNTER).increment();
+
+        return message.ack();
+    }
```

Comment:
I'm going to reopen this one too, because I am thinking on moving the transformation logic to yet another class too, and maybe wrap those two in a service:

- `ExportEventsService#extractFiltersAndTransform` which would make use of the filters extractor and the transformer.

That way the handler would only call that and deal with potential exceptions, and then send the proper response to the service.

Reply:
Done! I moved the filter extraction bit to a class, and then created a service which calls the different transformators. That should make the handler easier to follow.